src/Form/LegacyConsumer/Commands/SetupFieldPersistence.php

```diff
@@ -79,13 +80,13 @@ public function process( Field $field ) {
 					implode( '| ', array_values( array_filter( $data ) ) ) :
 					$data;
 
-				/** @var StoreAsMeta $field */
+				/** @var Text $field */
```

Comment:
This isn't too important, but what I’ve been doing for these docblock `@var` statements is thinking: if the case this is being used is more general, then I try to use a union type where `Field` is used as the base, but then also any necessary traits are also listed (e.g. `/* @var Field|StoreAsMeta */`). When it's more specific code, for example like when there is actual logic setup to only let `File` types through, then I’ll add a comment overriding the type of the variable to the type of field it actually is. Not sure if that resolves all of the IDE errors or warnings for you, but it does for me and the comment proves to be more accurate of what is actually represented in the code. I think it would be good to not use a specific type like this here just because it satisfies the IDE, since it leads to confusion for the human reading the code (the `$field` here isn’t only ever a `Text` type).

Reply:
In either case, this is hacky and the real code would break if the `Field` did not implement `StoreAsMeta`, which the editor is trying to point out. The real solution would be to create an interface which fields could implmenent. This way we could gate these functions using an `instanceof` statement. That’s a larger decision though and I think we should avoid it for now. Let’s just take note of this case and we can use it when or if we ever need to propose adding the interfaces in the future.